backends/xsmm/ceed-xsmm.h

```diff
@@ -17,12 +17,17 @@
 #include <ceed-backend.h>
 #include <libxsmm.h>
 #include <string.h>
+#include <khash.h>
 #include <math.h>
 
+// Instantiate khash structs and methods
+KHASH_MAP_INIT_INT(m32, char)
```

Comment:
You can make a map from tuple (as a struct) to kernel, and thus get rid of a lot of bookkeeping that you're doing manually. You'd call `KHASH_INIT(hashx, KeyTuple, ValueType, 1, KeyHash, KeyEqual)` where you write `KeyHash` and `KeyEqual`. There are examples of such functions in PETSc, e.g., https://gitlab.com/petsc/petsc/-/blob/master/src/dm/impls/plex/plexinterpolate.c#L9-33

Reply:
How about this? I largely adapted the PETSc macros (and called out the fact in the ceed-hash.h)